tools/prerequisites/k8s-setup.sh

```diff
@@ -121,6 +121,13 @@ install_kubernetes() {
 }
 
 initialize_cluster() {
+    if [ -d "$HOME/.kube" ]; then
+        rm -rf "$HOME/.kube"
+    fi
+
+    read -rp "Setting up single node cluster? (yes/no, default: no): " single_node_setup
+    single_node_setup=${single_node_setup:-"no"}
+
     read -rp "Enter Pod Network CIDR (default: 10.244.0.0/16): " pod_network_cidr
     pod_network_cidr=${pod_network_cidr:-"10.244.0.0/16"}
     command="kubeadm init --pod-network-cidr=${pod_network_cidr}"
```

Comment:
It would be better to specify the same `--kubernetes-version` as the installed version during init.

Reply:
I think the behavior of kubeadm init will depend on the version of kubeadm installed on the system.

Reply:
I try to install version 1.28.2 and then `kubeadm init`, it will use version 1.28.4:
```
kubeadm init
[init] Using Kubernetes version: v1.28.4
[preflight] Running pre-flight checks
```

Reply:
After check:
1. Kubernetes follows a versioning scheme of MAJOR.MINOR.PATCH. In this case, both 1.28.2 and 1.28.4 are part of the 1.28 minor version series, and kubeadm will default apply the latest patch version to fix bugs/issues and it won't bring any breaking changes.
2. kubeadm maintains compatibility across patch versions within a minor version. 

If we define the version with the kubeadm installation version, and user only execute this step, it may casue some issue,

Reply:
A case is that, user has installed 1.27.x, and only want to execute the cluster init, if we define the version which is different minor version, will casue issues.
Two solution:
1.When init cluster, we first check the installed kubeadm version and set it as default.
2.When init cluster, it will default using kubeadm to get the latest patch version and the minor is the same with kubeadm version,
```
kubeadm init
I1116 16:53:44.950609   16959 version.go:255] remote version is much newer: v1.28.4; falling back to: stable-1.23
[init] Using Kubernetes version: v1.23.17

```

Reply:
It makes senses